gradient/api_sdk/repositories/common.py

```diff
@@ -377,6 +377,84 @@ def _format_datetime(self, some_datetime):
         datetime_str = some_datetime.strftime("%Y-%m-%dT%H:%M:%SZ")
         return datetime_str
 
+class ListMetrics(GetResource):
+    OBJECT_TYPE = None
+
+    DEFAULT_INTERVAL = "30s"
+
+    @abc.abstractmethod
+    def _get_instance_by_id(self, instance_id, **kwargs):
+        pass
+
+    def _get_metrics_api_url(self, instance, protocol="https"):
+        if not instance.metrics_url:
+            raise GradientSdkError("Metrics API url not found")
+
+        metrics_api_url = concatenate_urls(protocol + "://", instance.metrics_url)
+        return metrics_api_url
+
+    def _get(self, **kwargs):
+        new_kwargs = self._get_kwargs(kwargs)
+        rv = super(ListMetrics, self)._get(**new_kwargs)
+        return rv
+
+    def _get_kwargs(self, kwargs):
+        instance_id = kwargs["id"]
+        instance = self._get_instance_by_id(instance_id)
+        started_date = self._get_start_date(instance, kwargs)
+        end = self._get_end_date(instance, kwargs)
+        interval = kwargs.get("interval") or self.DEFAULT_INTERVAL
+        metrics_api_url = self._get_metrics_api_url(instance)
+        new_kwargs = {
+            "start": started_date,
+            "interval": interval,
+            "objecttype": self.OBJECT_TYPE,
+            "handle": instance_id,
+            "metrics_api_url": metrics_api_url if config.CONFIG_HOST != "http://localhost:3102" else "% s:8080"% metrics_api_url,
+        }
+        if end:
+            new_kwargs["end"] = end
+
+        return new_kwargs
+
+    def get_request_url(self, **kwargs):
+        return "metrics/api/v1/list"
+
+    def _get_api_url(self, **kwargs):
+        api_url = kwargs["metrics_api_url"]
+        return api_url
+
+    def _get_start_date(self, instance, kwargs):
+        datetime_string = kwargs.get("start") or instance.dt_started or instance.dt_created
+        if not datetime_string:
+            return None
+
+        datetime_string = self._format_datetime(datetime_string)
+        return datetime_string
+
+    def _get_end_date(self, instance, kwargs):
+        datetime_string = kwargs.get("end")
+        if not datetime_string:
+            return None
+
+        datetime_string = self._format_datetime(datetime_string)
+        return datetime_string
```

Comment:
Does this work for both finished and ongoing workloads?

Reply:
yep confirmed with a running deployment and stopped the same deployment

still can't run experiments or jobs in psdev staging (psdev local fails immediately), but since this is a shared function, should work the same if the experiments or jobs were to successfully run